src/ActionDialog/ActionDialog.js

```diff
@@ -0,0 +1,121 @@
+import * as React from "react";
+
+import {
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  IconButton,
+  Stack
+} from "@mui/material";
+
+import CloseIcon from "@mui/icons-material/Close";
+import PropTypes from "prop-types";
+import { styled } from "@mui/material/styles";
+
+/**
+ * Horizontal alignment button group component to toggle between left/center/right/justify alignment
+ */
+export default function ActionDialog({
+  onCancelClick = () => {},
+  onSaveClick = () => {},
+  title = "Some title",
+  cancelText = "cancel",
+  saveText = "Save",
+  open = true,
+  content,
+  saveEnabled = false,
+  width = "400px",
+  showCloseIcon = true
+}) {
+  // action dialog styles
+  const ActionDialogWrapper = styled(Dialog)(({ theme }) => ({
+    "& .MuiDialog-container": {
+      "& .MuiPaper-root": {
+        maxWidth: width,
+        width: "100%"
+      }
+    },
+    "& .MuiDialogActions-root": {
+      padding: theme.spacing(2, 3)
+    }
+  }));
+  // return components
+  return (
+    <ActionDialogWrapper open={open} data-testid="action-dialog">
+      <DialogTitle fontWeight={600}>
+        {title}
+        {showCloseIcon ? (
+          <IconButton
+            data-testid="close-icon"
+            aria-label="close"
+            onClick={onCancelClick}
+            sx={{
+              color: theme => theme.palette.grey[500],
+              position: "absolute",
+              right: 8,
+              top: 8
+            }}
+          >
+            <CloseIcon />
+          </IconButton>
+        ) : null}
+      </DialogTitle>
+      <DialogContent dividers>
+        <Stack spacing={3}>{content}</Stack>
+      </DialogContent>
+      <DialogActions>
+        <Button onClick={onCancelClick}>{cancelText}</Button>
+        <Button
+          variant="contained"
+          onClick={onSaveClick}
+          disabled={saveEnabled}
+        >
+          {saveText}
+        </Button>
+      </DialogActions>
+    </ActionDialogWrapper>
+  );
+}
+
+ActionDialog.propTypes = {
```

Comment:
The proptypes here are not complete. 

Missing title, cancelText, saveText and open

Reply:
Added